apphub/image_classification/pyramidnet/pyramidnet.ipynb

```diff
@@ -0,0 +1,327 @@
+    "\n",
+    "import tensorflow as tf\n",
+    "import tensorflow_addons as tfa\n",
+    "from tensorflow.keras import layers\n",
+    "\n",
+    "import fastestimator as fe\n",
+    "from fastestimator.dataset.data.cifar10 import load_data\n",
+    "from fastestimator.op.numpyop.meta import Sometimes\n",
+    "from fastestimator.op.numpyop.multivariate import HorizontalFlip, PadIfNeeded, RandomCrop\n",
+    "from fastestimator.op.numpyop.univariate import CoarseDropout, Normalize\n",
+    "from fastestimator.op.tensorop.loss import CrossEntropy\n",
+    "from fastestimator.op.tensorop.model import ModelOp, UpdateOp\n",
+    "from fastestimator.trace.adapt import LRScheduler\n",
+    "from fastestimator.trace.io import BestModelSaver\n",
+    "from fastestimator.trace.metric import Accuracy\n",
+    "from fastestimator.util import get_num_devices"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "#training parameters\n",
+    "epochs = 150\n",
+    "batch_size = 32\n",
+    "max_train_steps_per_epoch = None\n",
+    "max_eval_steps_per_epoch = None\n",
+    "save_dir = tempfile.mkdtemp()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset Loading\n",
+    "\n",
+    "In this application, we apply the network on Cifar10 dataset as suggested by the paper."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data, eval_data = load_data()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preprocessing\n",
+    "The image pixel values are first normalized by predefined constant, then we apply random padding and cropping, followed by horizontal flip and cut out during training. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline = fe.Pipeline(\n",
+    "    train_data=train_data,\n",
+    "    eval_data=eval_data,\n",
+    "    batch_size=batch_size * get_num_devices(),\n",
+    "    ops=[\n",
+    "        Normalize(inputs=\"x\", outputs=\"x\", mean=(0.4914, 0.4822, 0.4465), std=(0.2471, 0.2435, 0.2616)),\n",
+    "        PadIfNeeded(min_height=40, min_width=40, image_in=\"x\", image_out=\"x\", mode=\"train\"),\n",
+    "        RandomCrop(32, 32, image_in=\"x\", image_out=\"x\", mode=\"train\"),\n",
+    "        Sometimes(HorizontalFlip(image_in=\"x\", image_out=\"x\", mode=\"train\")),\n",
+    "        CoarseDropout(inputs=\"x\", outputs=\"x\", mode=\"train\", max_holes=1)\n",
+    "    ])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize Preprocessing Results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [
+    {
+     "data": {
```

Comment:
this is an excessive number of example images

Reply:
done

Reply:
whatever happened is preventing the notebook from rendering anymore